drink/src/lib.rs

```diff
@@ -17,36 +17,51 @@ use std::sync::{Arc, Mutex};
 pub use bundle::ContractBundle;
 pub use drink_test_macro::{contract_bundle_provider, test};
 pub use errors::Error;
+use frame_support::traits::fungible::Inspect;
 pub use frame_support::{
     sp_runtime::{AccountId32, DispatchError},
     weights::Weight,
 };
 use frame_system::EventRecord;
 pub use mock::{mock_message, ContractMock, MessageMock, MockedCallResult, Selector};
-use pallet_contracts::{debug::ExecResult, ExecReturnValue};
+use pallet_contracts::{
+    debug::ExecResult, ContractExecResult, ContractInstantiateResult, ExecReturnValue,
+};
 use pallet_contracts_uapi::ReturnFlags;
 use parity_scale_codec::{Decode, Encode};
 /// Export pallets that are used in the minimal runtime.
 pub use {frame_support, frame_system, pallet_balances, pallet_contracts, pallet_timestamp};
 
 pub use crate::runtime::minimal::{self, MinimalRuntime};
 use crate::{
-    errors::MessageResult, mock::MockRegistry,
-    runtime::pallet_contracts_debugging::InterceptingExtT,
+    errors::MessageResult,
+    mock::MockRegistry,
+    runtime::{pallet_contracts_debugging::InterceptingExtT, AccountIdFor},
 };
 
 /// Alias for `frame-system`'s `RuntimeCall` type.
-pub type RuntimeCall<R> = <R as frame_system::Config>::RuntimeCall;
+pub type RuntimeCall<Runtime> = <Runtime as frame_system::Config>::RuntimeCall;
 
 /// Alias for `pallet-balances`'s Balance type.
-pub type BalanceOf<R> = <R as pallet_balances::Config>::Balance;
+pub type BalanceOf<Runtime> =
```

Comment:
the difference is from which pallet we assume is present... and in most places where we need this type alias we are working with just contracts pallet :| but I know it's not perfect :disappointed:

Reply:
My undestanding was that `Balance` type is "defined" by the `pallet_balances` - not `pallet_contracts`. That's why on Aleph Zero `Balance` has 12 digits and on other chains it might have a different number. 

Is `pallet_contracts`' `Balance` type somehow "tied" to that from `pallet_balances`? Or, differently, can it be the case that `pallet_contracts`'s `Balance` will differ from the one in `pallet_balances`?

Reply:
it depends on the runtime configuration (i.e. is tied there); in 99.9% this will be the same type, but using the balance from `pallet_contracts` is, say, more abstract